src/syncState.js

```diff
@@ -93,10 +93,11 @@ export function MessageListener({ channel, dispatch, allowed }) {
     this.messageChannel.onmessage = this.handleOnMessage;
 }
 
-export const createStateSyncMiddleware = (config = defaultConfig) => {
-    const allowed = isActionAllowed(config);
-    const channel = new BroadcastChannel(config.channel, config.broadcastChannelOption);
-    const prepareState = config.prepareState || defaultConfig.prepareState;
+export const createStateSyncMiddleware = partialConfig => {
+    const completeConfig = { ...defaultConfig, ...partialConfig };
```

Comment:
I think technically you'd want to remove `undefined` from `partialConfig` before spreading it here:

![image](https://user-images.githubusercontent.com/3213339/172908376-7f2bccb6-01c2-4899-855f-2db61e7ca47d.png)